skimage/feature/_kadir_brady.py

```diff
@@ -0,0 +1,325 @@
+
+    Returns
+    -------
+    A : (3, N) ndarray of float
+        A 2d array with each column representing 3 values, '(y, x, scale)'
+        where '(y,x)' are coordinates of the region and 'scale' is the
+        size of corresponding salient region.
+    """
+
+    t_gamma, t_scale, t_r, t_c, D = _salient_regions(candidate_regions, saliency_threshold)
+    gamma, scale, row, column = (np.array([]) for i in range(4))
+    
+    n = max(t_gamma.shape)
+    n_reg = 0
+    # clusters matrix
+    cluster = np.zeros((3, k + 1))
+
+    # pruning process
+    for index in range(n):
+        cluster[0, 0] = t_c[index]
+        cluster[1, 0] = t_r[index]
+        cluster[2, 0] = t_scale[index]
+        s_i = np.argsort(D[index, :])
+        # fill in the neighbouring regions
+        for j in range(k):
+            cluster[0, j+1] = t_c[s_i[j+1]]
+            cluster[1, j+1] = t_r[s_i[j+1]]
+            cluster[2, j+1] = t_scale[s_i[j+1]]
+
+        # clusters center point
+        center = np.array([np.mean(cluster, axis=1)])
+
+        # check if the regions are "suffiently clustered", if variance is less than threshold
+        v = np.var(np.sqrt(((cluster - np.tile(center.T, (1, k + 1))) ** 2).sum(axis=0)))
+        if v > v_th:
+            continue
+
+        center = np.mean(cluster, axis=1)
+        if n_reg > 0:
+            # make sure the region is "far enough" from already clustered regions
+            d = np.sqrt(((np.array(list(zip(column, row, scale)))
+                            - np.tile(center.T, (n_reg, 1))) ** 2).sum(axis=1))
+            if (center[2] >= 0.6*d).sum() == 0:
+                n_reg = n_reg + 1
+                column = np.append(column, center[0])
+                row = np.append(row, center[1])
+                scale = np.append(scale, center[2])
+                gamma = np.append(gamma, t_gamma[index])
+        else:
+            n_reg = n_reg + 1
+            column = np.append(column, center[0])
+            row = np.append(row, center[1])
+            scale = np.append(scale, center[2])
+            gamma = np.append(gamma, t_gamma[index])
+
+    return np.array([row, column, scale])
+
+
+def _saliency_param(image, scales):
+    """Calculate parameters for Saliency metric.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input grayscale image
+    scales : ndarray
+        Scales for key regions
+
+    Returns
+    -------
+    entropy : (M, N) ndarray of float
+        Matrix containing entropy value of region over scales.
+    weights : (M, N) ndarray of float
+        Matrix containing abs difference of PDF over scales.
+    r : (N,) ndarray of float
+        Row index of regions.
+    c : (N,) ndarray of float
+        Column index of regions.
+    """
+    
+    image = img_as_ubyte(image)
+    nr, nc = image.shape
+    # find pixels that we are going to examine
+    mask = np.ones((nr - max(scales), nc - max(scales)))
+    r, c = np.nonzero(mask) + max(scales) / 2 + 1
```

Comment:
r and c are assigned here the 2 arrays inside the tuple, so it's actually array+float
```
In [96]: np.nonzero(mask)
Out[96]: (array([0, 0, 1, 1]), array([0, 1, 0, 1]))

In [97]: r, c = np.nonzero(mask)

In [98]: r
Out[98]: array([0, 0, 1, 1])

In [99]: c
Out[99]: array([0, 1, 0, 1])
```